code/obj/artifacts/artifact_items/key.dm

```diff
@@ -0,0 +1,352 @@
+				door.outer_entrance_spawned = TRUE
+			if (EAST_ENTRANCE)
+				var/obj/cross_dummy/west/w = new (entrance, src.backroom_region.turf_at(17, 19))
+				var/obj/cross_dummy/east/e = new (src.backroom_region.turf_at(18, 29), get_step(entrance, EAST))
+				src.east_dummies += w
+				src.east_dummies += e
+				var/obj/artifact_door/door = locate() in src.backroom_region.turf_at(17, 19)
+				door.outer_entrance_spawned = TRUE
+			if (WEST_ENTRANCE)
+				var/obj/cross_dummy/east/e = new (entrance, src.backroom_region.turf_at(13, 19))
+				var/obj/cross_dummy/west/w = new (src.backroom_region.turf_at(12, 19), get_step(entrance, WEST))
+				src.west_dummies += e
+				src.west_dummies += w
+				var/obj/artifact_door/door = locate() in src.backroom_region.turf_at(13, 19)
+				door.outer_entrance_spawned = TRUE
+
+		entrance.density = FALSE
+		var/obj/artifact_door/wooden_door = new(entrance)
+		wooden_door.set_dir(get_dir(wooden_door, user))
+		src.update_visual_mirrors(entrance, entrance_dir)
+		entrance.icon = 'icons/turf/floors.dmi'
+		entrance.icon_state = "darkvoid"
+		entrance.opacity = TRUE
+		entrance.name = "Thick void mist"
+		entrance.desc = "Void mist thick enough that you can't see through it.. How did this get here?"
+		RL_UPDATE_LIGHT(entrance)
+
+	proc/update_visual_mirrors(turf/station_reference, entrance)
+		var/col_start
+		var/col_end
+		var/row_start
+		var/row_end
+		var/x_start_offset
+		var/y_start_offset
+
+		switch(entrance)
+			if (NORTH_ENTRANCE)
+				col_start = 1
+				col_end = 29
+				row_start = 25
+				row_end = 35
+				x_start_offset = -15
+				y_start_offset = -24
+			if (SOUTH_ENTRANCE)
+				col_start = 1
+				col_end = 29
+				row_start = 1
+				row_end = 13
+				x_start_offset = -15
+				y_start_offset = -14
+			if (EAST_ENTRANCE)
+				col_start = 18
+				col_end = 29
+				row_start = 1
+				row_end = 35
+				x_start_offset = -17
+				y_start_offset = -19
+			if (WEST_ENTRANCE)
+				col_start = 1
+				col_end = 12
+				row_start = 1
+				row_end = 35
+				x_start_offset = -13
+				y_start_offset = -19
+
+		var/turf/T
+		var/turf/station_turf
+		for (var/i = col_start to col_end)
+			for (var/j = row_start to row_end)
+				T = src.backroom_region.turf_at(i, j)
+				station_turf = locate(station_reference.x + x_start_offset + i, station_reference.y + y_start_offset + j, station_reference.z)
+
+				if (station_turf)
+					station_turf.appearance_flags |= KEEP_TOGETHER
+					T.vis_contents = null // clear previously assigned vis_contents
+					//RL_UPDATE_LIGHT(T)
+					//T.RL_AddOverlay()
+					//RL_APPLY_LIGHT(T, null, null, station_turf.RL_GetBrightness(), 0, 255, 255, 255)
+					T.vis_contents += station_turf
+					T.density = station_turf.density
+					T.opacity = station_turf.opacity
+					T.name = station_turf.name
+					T.desc = station_turf.desc
+					T.icon = station_turf.icon
+					T.icon_state = station_turf.icon_state
+					//T.fullbright = TRUE
+					T.RL_Init()
+
+/datum/artifact/key
```

Comment:
Might be better to rename this to be "dimensional_key" as to not confuse it with "activator_key".

Reply:
Yeah these are more temporary names, gonna give them better names before final draft